runtime/src/main/java/org/corfudb/runtime/view/LayoutBuilder.java

```diff
@@ -295,7 +291,11 @@ public LayoutBuilder assignResponsiveSequencerAsPrimary(Set<String> endpoints) {
         List<String> modifiedSequencerServers = new ArrayList<>(layout.getSequencers());
         for (int i = 0; i < modifiedSequencerServers.size(); i++) {
             String sequencerServer = modifiedSequencerServers.get(i);
-            if (!endpoints.contains(sequencerServer)) {
+
+            // Add a sequencer server given sequencerServer is not already included in
+            // endpoints AND it is not included in the unresponsive nodes of the layout.
+            if (!endpoints.contains(sequencerServer) &&
```

Comment:
I think we need a log warning or error when we cannot choose a sequencer ? It is hard to debug if we have a problem in this area.

Reply:
Thanks for the good point. Added a warning log with relevant information.